src/ast.fs

```diff
@@ -59,7 +59,7 @@ and TypeSpec =
 
 and Type = {
     name: TypeSpec // e.g. int
-    typeQ: string option // e.g. const, uniform
+    typeQ: string list option // e.g. const, uniform
```

Comment:
Couldn't this be just `string list`?
I suppose both `None` and `[]` mean "no qualifiers".

Reply:
Indeed it can, thanks -- the formatter actually treats `None` and `[]` slightly differently right now (where/how spaces get inserted), hence the code I put somewhere that converts from `[]` to `None` and ugh yes just using `string list` I think will be a lot better.